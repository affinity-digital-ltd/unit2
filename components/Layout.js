import { Component } from 'react'

import Link from 'next/link'

import '../scss/application.scss'

class Layout extends Component {
  constructor (props) {
    super(props)

    this.state = {
      navClass: '',
      navBrandClass: ''
    }
  }

  componentDidMount () {
    if (this.props.url.pathname === '/') {
      this.setState({
        navClass: 'c-nav',
        navBrandClass: 'c-nav__brand'
      })
    }
  }

  render () {
    return <div>
      <nav className={`navbar navbar-expand-lg navbar-light bg-light ${this.state.navClass}`}>
        <Link href='/'><a className={`navbar-brand ${this.state.navBrandClass}`}>Unit 1</a></Link>
      </nav>
      <div className='container'>
        { this.props.children }
      </div>
    </div>
  }
}

export default Layout
