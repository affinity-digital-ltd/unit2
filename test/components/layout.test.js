import Layout from '../../components/Layout'

describe('<Layout />', () => {
  it('it renders correctly', () => {
    const url = {
      pathname: '/'
    }
    
    const component = shallow(
      <Layout url={url}/>
    )

    expect(component).toMatchSnapshot()
  })
})
