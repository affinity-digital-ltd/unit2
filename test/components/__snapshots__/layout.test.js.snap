// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> it renders correctly 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light c-nav"
  >
    <Link
      href="/"
    >
      <a
        className="navbar-brand c-nav__brand"
      >
        Unit 1
      </a>
    </Link>
  </nav>
  <div
    className="container"
  />
</div>
`;
